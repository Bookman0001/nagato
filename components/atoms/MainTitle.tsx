import styled from 'styled-components'

interface Props {
  children: React.ReactNode
}

export default function MainTitle({ children }: Props) {
  return (
    <TitleContainer>
      <>{children}</>
    </TitleContainer>
  )
}

const TitleContainer = styled.h1`
  font-size: 32px;
  span {
    font-size: 64px;
  }
`
