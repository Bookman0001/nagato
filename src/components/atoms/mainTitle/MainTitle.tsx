import React from 'react'
import styled from 'styled-components'

import { DEVICE_WIDTH, FONT_SIZE } from 'src/theme/constants'

interface Props {
  children: React.ReactNode
}

export function MainTitle({ children }: Props) {
  return <TitleContainer>{children}</TitleContainer>
}

const TitleContainer = styled.h1`
  display: flex;
  justify-content: center;
  align-items: center;
  line-height: 1.5;
  font-size: ${FONT_SIZE.L};
  padding-bottom: 4rem;
  @media (max-width: ${DEVICE_WIDTH.PHONE}) {
    font-size: ${FONT_SIZE.M};
  }
`
