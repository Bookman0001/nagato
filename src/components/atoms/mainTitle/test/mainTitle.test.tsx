import { shallow } from 'enzyme'

import MainTitle from '../'

test('MainTitle', () => {
  const component = shallow(<MainTitle>test</MainTitle>)
  expect(component).toMatchSnapshot()
})
