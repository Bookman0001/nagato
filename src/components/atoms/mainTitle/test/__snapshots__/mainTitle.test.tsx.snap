// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTitle 1`] = `
<styled.h1>
  test
</styled.h1>
`;
